networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/libp2p/LibP2PPeer.java

```diff
@@ -58,11 +65,29 @@ public boolean isConnected() {
 
   @Override
   @SuppressWarnings("FutureReturnValueIgnored")
-  public void disconnect() {
+  public void disconnectImmediately() {
     connected.set(false);
     connection.close();
   }
 
+  @Override
+  public void disconnectCleanly(final DisconnectRequestHandler.DisconnectReason reason) {
+    connected.set(false);
```

Comment:
It seems like we're calling `connected.set(false)` twice. Is there a specific reason we do that?

Reply:
We want to mark the peer as disconnected as soon as we start the clean disconnect process, but also want to do so when we disconnect immediately.  It doesn't make a lot of difference but just avoids us sending a request for blocks to a peer that we've started disconnecting.